Applications/VpView/vpViewCore.cxx

```diff
@@ -5683,58 +5654,26 @@ void vpViewCore::createDisplayOutline(vpProject* project)
     }
   else if (this->UseRawImageCoordinates)
     {
-    if (project->AOILowerRightLatLon[0] != 444)
-      {
-      double lr[4] = { project->AOILowerRightLatLon[1],
-                       project->AOILowerRightLatLon[0], 0, 1.0 };
-
-      this->LatLonToWorldMatrix->MultiplyPoint(lr, lr);
-      if (lr[3] != 0)
-        {
-        lr[0] /= lr[3];
-        lr[1] /= lr[3];
-        }
-
-      if (project->AOILowerLeftLatLon[0] != 444 &&
-          project->AOIUpperRightLatLon[0] != 444)
-        {
-        // All four corners.
-        double ll[4] = { project->AOILowerLeftLatLon[1],
-                         project->AOILowerLeftLatLon[0], 0, 1.0 };
-
-        this->LatLonToWorldMatrix->MultiplyPoint(ll, ll);
-        if (ll[3] != 0)
-          {
-          ll[0] /= ll[3];
-          ll[1] /= ll[3];
-          }
-        double ur[4] = { project->AOIUpperRightLatLon[1],
-                         project->AOIUpperRightLatLon[0], 0, 1.0 };
-
-        this->LatLonToWorldMatrix->MultiplyPoint(ur, ur);
-        if (ur[3] != 0)
-          {
-          ur[0] /= ur[3];
-          ur[1] /= ur[3];
-          }
-
-        aoiOutlinePoints->SetPoint(0, ll[0], ll[1], 0);
-        aoiOutlinePoints->SetPoint(1, lr[0], lr[1], 0);
-        aoiOutlinePoints->SetPoint(2, ur[0], ur[1], 0);
-        aoiOutlinePoints->SetPoint(3, 0, 0, 0);
-        }
-      else
-        {
-        // Two corners.
-        aoiOutlinePoints->SetPoint(0, 0, lr[1], 0);
-        aoiOutlinePoints->SetPoint(1, lr[0], lr[1], 0);
-        aoiOutlinePoints->SetPoint(2, lr[0], 0, 0);
-        aoiOutlinePoints->SetPoint(3, 0, 0, 0);
-        }
+    if (project->AOI.GCS > 0)
+      {
+      const auto& aoi =
+        vgGeodesy::convertGcs(project->AOI, vgGeodesy::LatLon_Wgs84);
+      const auto& corners = aoi.Coordinate;
+      double ur[3] = {corners[1].Easting, corners[1].Northing, 0.0};
+      double lr[3] = {corners[2].Easting, corners[2].Northing, 0.0};
+      double ll[3] = {corners[3].Easting, corners[3].Northing, 0.0};
+      vtkVgApplyHomography(ur, this->LatLonToWorldMatrix, ur);
+      vtkVgApplyHomography(lr, this->LatLonToWorldMatrix, lr);
+      vtkVgApplyHomography(ll, this->LatLonToWorldMatrix, ll);
+
+      aoiOutlinePoints->SetPoint(0, ll);
+      aoiOutlinePoints->SetPoint(1, lr);
+      aoiOutlinePoints->SetPoint(2, ur);
+      aoiOutlinePoints->SetPoint(3, 0, 0, 0);
```

Comment:
Previously supported two corners (or four corners), but now only supports four corners?

Reply:
Hmm, I'm not sure how the VidTK reader is affected (possibly going to become a moot point), but the QSettings reader handles this, fairly robustly, at the parser level.

Reply:
This is "only" for the "display outline". Still a change in functionality, so I'm a bit uneasy about it, but don't want to get hung up on this - another case of very old code (2011 / 2012).